### ISSUE TITLE
DatePicker: Accessibility issues

### ISSUE THREAD

**Issue by pfumi** [primefaces/primefaces#13846]
### Describe the bug

axe-core raises accessiblity issues for the DatePicker component.

**Issue 1**
![Image](https://github.com/user-attachments/assets/23166eac-d1f1-4b9e-a5ef-8d807c7eb22d)

**Issue 2**
![Image](https://github.com/user-attachments/assets/22e9b4c2-6628-45fa-929a-1ded4cbd488a)

### Reproducer

1. Run [primefaces-test-issue13846.zip](https://github.com/user-attachments/files/20443887/primefaces-test-issue13846.zip)
2. Click into the datepicker field
3. Issues arise mentioned in the "Describe the bug" section

### Expected behavior

Issue 1: "aria-selected" should be used in a correct manner. "aria-selected" is not permitted on a link element.
A solution could be to set role="gridcell" on the `<td>`, and set the "aria-selected" on the `<td>` instead of the `<a>`.
An other solution is to change the `<a>` to `<button>` and use "aria-pressed".

Issue 2: The cell is only disabled styling wise, it should also set an ARIA attribute like "aria-disabled". E.g. Primevue does that -> https://primevue.org/datepicker/

### PrimeFaces edition

Community

### PrimeFaces version

15.0.3

### JSF implementation

All

### JSF version

4.0

### Java version

11

**Comment by pfumi** [comment]
I looked at other ui component libraries, and the solution with `role="gridcell"` and `aria-selected="true"` on the `<td>` tag is pretty common. I would suggest to only set the "true" case, and if !true to remove the attribute all along.

**Comment by melloware** [comment]
OK not a bad idea.

**Comment by melloware** [comment]
@pfumi 15.0.5-SNAPSHOT showcase is here: https://primefaces15-showcase-f233g.ondigitalocean.app/index.xhtml

Can you test all these accessibility fixes we made in the last week to make sure it all looks good?

**Comment by pfumi** [comment]
I'm on vacation at the moment, coming back at Tuesday. Then I will have a look.

**Comment by pfumi** [comment]
testet and looks fine 👍
